### ISSUE TITLE
Enhance the modeling!

### ISSUE THREAD

**Issue by matthew-sochor** [matthew-sochor/fish.io.ai#10]
@MattGraz lets interface a bit here with what we are both doing

As per @thenomemac 's advice, resnet50 and inceptionv3 are smaller and faster for cpu.  So lets split those up.

I'll try and optimize resnet50 if you take inception and in the end we can just do an ensemble model or something?  We can post updates here as well with tips and tricks

**Comment by matthew-sochor** [comment]
Resnet50 results for Walleye v Carp

1. Preprocess with top=False, 2x 256 dense layers :
loss: 0.0362 - acc: 0.9898 - val_loss: 0.6820 - val_acc: 0.9133

2. Preprocess with top=False, 1024 - 512 - 256 layers:
loss: 0.0312 - acc: 0.9936 - val_loss: 0.7449 - val_acc: 0.9296

**Comment by MattGraz** [comment]
Snap, that's awesome

**Comment by matthew-sochor** [comment]
yes, slightly modified (i.e. changed vgg for resnet and played with layers a bit) but otherwise its the same

**Comment by MattGraz** [comment]
Cool.  I'm playing with Inception right now (Added BatchNorm and adjusted dropout()).  Getting about 85% accuracy on val and loss of ~ .23.  I have yet to play with any transformed batches.  I'll run that overnight and have some result to share tomorrow.  

It looks like we have something pretty decent for C vs. W.  I'll probably try messing around with a 4 class model tomorrow and see if we can get it to generalize still.

**Comment by thenomemac** [comment]
We can talk about this on Tuesday but we may be able to do even better
using additional convolutions at end of research net stack.

**Comment by thenomemac** [comment]
@matthew-sochor can you publish the code for your resent-50 93% acc model. I'd might play with that tonight and see if I can't do better with some hyper parameter tweaks. Awesome progress, seems like a viable model.

**Comment by matthew-sochor** [comment]
@thenomemac published in PR #11 

The model weights are in the google drive here https://drive.google.com/file/d/0B0jBfw2ul1mBeVgzSVhKZFc4SGM/view?usp=sharing

**Comment by MattGraz** [comment]
FYI, the inception models I tried, the best was the first one listed above.  

Playing with batch size, batchnorm, and image rotation either deteriorated or equaled the previous results.  Let's stick with @matthew-sochor's work above.

**Comment by MattGraz** [comment]
Multinomial Model with updated perch dataset.  Very slight boost over what was previously run.  This "boost" is likely more due to not having as many difficult images in the dataset than it is due to the model actually learning anything.  

More to come. 

loss: 0.1816 - acc: 0.9180 - val_loss: 2.4060 - val_acc: 0.7538

**Comment by thenomemac** [comment]
Woot 98%

**Comment by thenomemac** [comment]
Good news is if we advance to next round we have more experiments to run!

**Comment by MattGraz** [comment]
We could always just stick withe the Carp vs Walleye model for round 1.  

Have you tried running your model with the updated white/yellow perch?  I went from ~72% to 75% from removing some more junk images.  I'd expect you could also see a slight jump up if you use that as well.

**Comment by thenomemac** [comment]
No, I'm not sure where the junk list is. I think what we have should still work fine for demo with high probability. We'll test it on the print-out tomorrow.

**Comment by thenomemac** [comment]
Once my pipeline is added to make I'm still very confident we can get to 95+ acc by scrapping more data and doing more robust hyper parameter search.

**Comment by thenomemac** [comment]
Think we're well on our way thanks to big accuracy leaps in pull #35

**Comment by matthew-sochor** [comment]
closing
